### PR TITLE
Removed guzzle3 dependency

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | is a part of #2106 
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR removes dependency to guzzle3 and replaces it with guzzlehttp.

### CODE REVIEW

**Comment by wachterjohannes** [comment]
i have found out that the cache component and websocket also uses guzzle

**Comment by wachterjohannes** [comment]
@danrot i have to look at this next week now the travis tests are stalling ...

**Comment by wachterjohannes** [comment]
@danrot tests are green now

File: src/Sulu/Component/HttpCache/ProxyClient/Symfony.php [hunk 1]
```diff
@@ -41,7 +42,7 @@ class Symfony implements ProxyClientInterface, PurgeInterface
     /**
      * Request queue.
      *
-     * @var array|RequestInterface[]
+     * @var array|Request[]
```
Comment (danrot) [inline src/Sulu/Component/HttpCache/ProxyClient/Symfony.php]: Why don't we remove the `array`? Is it not always an array of requests?

File: src/Sulu/Component/HttpCache/ProxyClient/Symfony.php [hunk 2]
```diff
@@ -102,79 +103,60 @@ protected function queueRequest($method, $url, array $headers = [])
...
-            } else {
-                // Unexpected exception type
-                $e = $exception;
-            }
-
-            $collection->add($e);
         }
 
-        throw $collection;
+        Promise\settle($promises);
```
Comment (danrot) [inline src/Sulu/Component/HttpCache/ProxyClient/Symfony.php]: Why don't we just use `settle` here?

Comment (wachterjohannes) [inline src/Sulu/Component/HttpCache/ProxyClient/Symfony.php]: settle is a function in the Promise namespace. functions can be imported by `use function` but this PHP5.6 http://php.net/manual/en/language.namespaces.importing.php

**Comment by wachterjohannes** [comment]
@danrot fixed

**Comment by danrot** [comment]
@wachterjohannes Can we tick of an item of #2106 now?

**Comment by wachterjohannes** [comment]
yes we can and i did (:

**Comment by yukoff** [comment]
There are still some leftovers - guzzle/plugin-mock in some places and at least cboden/ratchet <0.4 depends on guzzle3 :(

**Comment by wachterjohannes** [comment]
@yukoff yes we have the dependency over `cboden/ratchet` the problem there is that there is no progess since a long time ...

**Comment by yukoff** [comment]
@wachterjohannes Yeah :( But 0.4 branch got some commits recently and travis build is green ;) Not sure `^0.4@dev` is a good idea, but I think I'll try that out.

**Comment by wachterjohannes** [comment]
@yukoff some changes has to be made in `src/Sulu/Component/Websocket` and `src/Sulu/Bundle/WebsocketBundle` where we use some parts of ratchet & guzzle3 which was returned from the websocket library.

If you have a working prototype we would love to see a PR from you.

**Comment by yukoff** [comment]
I think I'll get into this later today/tomorrow.

**Comment by yukoff** [comment]
Seems ready, created #3230 which in turn depends on #3229 as fos/http-cache <2.0 also depends on guzzle3 :(

**Pull request merged**